Improve get terminal window size in the docker container #15

skanehira opened the pull request:
relate #15

skanehira committed:
Improve get terminal window size in the docker container #15

glvr182 commented on gui.go:
```diff
@@ -70,7 +74,8 @@ type Gui struct {
 // NewGui returns a new Gui object with a given output mode.
 func NewGui(mode OutputMode) (*Gui, error) {
-	if err := termbox.Init(); err != nil {
+	err := termbox.Init()
```
Might not be necessary to split this up

skanehira replied:
I split it because I want to use the err in L90

glvr182 replied:
Ah okay, resolving this then

glvr182 commented on gui.go:
```diff
@@ -634,3 +642,44 @@ func (g *Gui) execKeybindings(v *View, ev *termbox.Event) (matched bool, err err
 	}
 	return matched, nil
 }
+
```
could you add some basic documentation to this function?

skanehira replied:
I added the documentation :)

glvr182 commented on gui.go:
```diff
@@ -634,3 +642,44 @@ func (g *Gui) execKeybindings(v *View, ev *termbox.Event) (matched bool, err err
+
+	var termw, termh int
+
+	out, err := os.OpenFile("/dev/tty", os.O_RDWR, 0)
```
This might prevent gocui from working under systems that are not linux (correct me if im wrong)

mjarkk replied:
Also if this check is just for checking if it's ran inside docker this will then never work in a native windows docker container.

skanehira replied:
Docker container as the Linux container so we just need to check and get the term window size for Linux.
However gocui support windows so There need to check runtime os.
I’ll fixe this logic.

mjarkk replied:
Also i'm not sure how much people use this library on BSD based OSes but it might be handy to support those too.

skanehira replied:
I fixed this logic.

glvr182 commented on gui.go:
```diff
@@ -634,3 +642,44 @@ func (g *Gui) execKeybindings(v *View, ev *termbox.Event) (matched bool, err err
+
+	for {
+		_, _, _ = syscall.Syscall(syscall.SYS_IOCTL,
+			out.Fd(), uintptr(syscall.TIOCGWINSZ), uintptr(unsafe.Pointer(&sz)))
```
Is there a method that does not require unsafe.Pointer?

skanehira replied:
Sorry. I don’t know the other way.
This way used in the termbox. I think this no problem.
https://github.com/nsf/termbox-go/blob/master/termbox.go#L150+L155

glvr182 reviewed (commented):
Here are some things I was curious about, overall looks good! Will approve after the questions are answered.

skanehira committed:
fixed some

- Add documents for new function
- Only inside the container to use original get terminal window function

glvr182 commented:
Looks good! Can you rebase on the master branch?

skanehira commented:
@glvr182 I rebased :)

glvr182 commented:
@skanehira Then i will merge this
